vpr/src/noc/negative_first_routing.h

```diff
@@ -0,0 +1,70 @@
+#ifndef VTR_NEGATIVE_FIRST_ROUTING_H
+#define VTR_NEGATIVE_FIRST_ROUTING_H
+
+/**
+ * @file
+ * @brief This file declares the NegativeFirstRouting class, which implements
+ * the negative-first routing algorithm.
+ *
+ * Overview
+ * ========
+ * The NegativeFirstRouting class performs packet routing between routers in the
+ * NoC using negative-first routing algorithm. In this algorithm, moving towards
+ * negative directions (west and south) has a higher priority. Once the traffic flow
+ * moves towards north or east, it is no longer allowed to take a turn that aligns
+ * the route towards south and west.
+ */
```

Comment:
To increase path diversity, our implementation also ... (describe probability of going N/S vs. E/W that you use, and why).